analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LockReleasedAllPaths.Monitor.TryEnter.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Threading;
+
+class Program
+{
+    private object obj = new object();
+
+    public void Method1()
+    {
+        if (Monitor.TryEnter(obj)) // FN
+        {
+        }
+        else
+        {
+            Monitor.Exit(obj);
+        }
+    }
+
+    public void Method2()
+    {
+        if (Monitor.TryEnter(obj)) // Compliant
+        {
+            Monitor.Exit(obj);
+        }
+        else
+        {
+        }
+    }
+
+    public void Method3()
+    {
+        if (Monitor.TryEnter(obj, 42)) // FN
+        {
+        }
+        else
+        {
+            Monitor.Exit(obj);
+        }
+    }
+
+    public void Method4(bool condition)
+    {
+        bool isAcquired = false;
+        Monitor.TryEnter(obj, 42, ref isAcquired);
+        if (condition)
+        {
+            Monitor.Exit(obj);
+        }
```

Comment:
This is FN, because its not always unlocked
```suggestion
        bool isAcquired = false;
        Monitor.TryEnter(obj, 42, ref isAcquired); // FN
        if (condition)
        {
            Monitor.Exit(obj);
        }
```

Reply:
Let's add compliant counterpart:
```
        bool isAcquired = false;
        Monitor.TryEnter(obj, 42, ref isAcquired);
        if (isAcquired)
        {
            Monitor.Exit(obj);
        }
```